components/ReadingChallenge/ReadingChallenge.js

```diff
@@ -0,0 +1,192 @@
+import styled from "styled-components";
+import { useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import ChallengeBookList from "../ChallengeBookList/ChallengeBookList";
+
+export default function ReadingChallenge({
+  books,
+  booksInfo,
+  handleToggleAlreadyRead,
+}) {
+  const [modalState, setModalState] = useState({
+    isOpen: false,
+    isCancelled: false,
+  });
+
+  function openModal() {
+    setModalState({ isOpen: true, isSaved: false });
+  }
+
+  function closeModal() {
+    setModalState({ isOpen: false, isSaved: false });
```

Comment:
Since `isSaved` doesn't change, you could do without this key, right? 
If that's the truth 🙂, you could use a simple toggle function that inverts the value of `isOpen`.

```js
function toggleModal() {
  setModalState((prev) => ({ isOpen: !prev.isOpen }));
}
```